src/main/java/io/nats/client/impl/NatsJetStreamMetaData.java

```diff
@@ -25,53 +25,108 @@
 
     private static final long NANO_FACTOR = 10_00_000_000;
 
+    private final String prefix;
+    private final String domain;
+    private final String accountHash;
     private final String stream;
     private final String consumer;
     private final long delivered;
     private final long streamSeq;
     private final long consumerSeq;
     private final ZonedDateTime timestamp;
-    private long pending = -1;
+    private final long pending;
+    private final String token;
 
     @Override
     public String toString() {
         return "NatsJetStreamMetaData{" +
-                "stream='" + stream + '\'' +
+                "prefix='" + prefix + '\'' +
+                "domain='" + domain + '\'' +
+                ", accountHash='" + accountHash + '\'' +
```

Comment:
If this is somehow public, then `accountHash` and `token` don't need to be exposed. This is just internals needing to properly route the message back. No need to expose those to the user.

Reply:
@kozlovic We should capture this in some ADR. Is there a security reason these should not be public?

Reply:
It is: https://github.com/nats-io/nats-architecture-and-design/pull/42/files#diff-f52a7740d240a96a07d8e7f9a40d3eb460cee1176914ea5ae1d76175f0d140bdR172.

There is no need for the user to know those, they are simply for routing. Making them public would just add to the "breaking changes" issues if one day we want to alter those.

Also take note regarding the token count check moving forward: https://github.com/nats-io/nats-architecture-and-design/pull/42/files#diff-f52a7740d240a96a07d8e7f9a40d3eb460cee1176914ea5ae1d76175f0d140bdR174

Reply:
ok will fix

Reply:
Fixed.
1. don't expose account hash and token publicly
2. allow more than 12 tokens for future compatibility